platform/android/src/style/sources/source.cpp

```diff
@@ -43,6 +43,10 @@ namespace android {
         return jni::Make<jni::String>(env, source.getID());
     }
 
+    jni::String Source::getAttribution(jni::JNIEnv& env) {
+        return source.getAttribution() ? jni::Make<jni::String>(env, source.getAttribution().value()) : jni::Make<jni::String>(env,"");
```

Comment:
Minor thing, but maybe invoke `source.getAttribution()` only once. Now it doesn't really do much, but if the implementation gets more complex it's a bit wasteful.